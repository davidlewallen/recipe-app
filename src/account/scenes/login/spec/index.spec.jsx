import React from 'react';
import { shallow } from 'enzyme';

import Login from '../components';

describe('Login component snapshot test', () => {
  const props = {
    username: 'testUsername',
    handleUsername: () => {},
    password: 'testPassword',
    handlePassword: () => {},
    error: { value: false, message: '' },
    login: () => {},
  };

  it('should match snapshot', () => {
    const component = shallow(<Login {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should display error message', () => {
    const component = shallow(<Login {...props} error={{ value: true, message: 'test error' }} />);

    expect(component).toMatchSnapshot();
  });
});
