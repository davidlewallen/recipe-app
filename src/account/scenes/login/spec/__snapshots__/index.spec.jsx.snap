// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component snapshot test should display error message 1`] = `
<Grid
  bsClass="container"
  className="login"
  componentClass="div"
  fluid={false}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={6}
      smOffset={3}
      xs={12}
    >
      <form>
        <FormGroup
          bsClass="form-group"
          className="align-center"
        >
          <p
            className="error-message"
          >
            test error
          </p>
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Username
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="username-input"
            componentClass="input"
            id="username"
            onChange={[Function]}
            placeholder="E.g. JohnDoe123"
            type="text"
            value="testUsername"
          />
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Password
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="password-input"
            componentClass="input"
            id="password"
            onChange={[Function]}
            placeholder="E.g. 124!Jigzx"
            type="password"
            value="testPassword"
          />
          <Button
            active={false}
            block={true}
            bsClass="btn"
            bsStyle="primary"
            className="login-button"
            disabled={false}
            onClick={[Function]}
          >
            Login
          </Button>
          <div
            className="register-text"
          >
            New to My Saved Recipes? 
            <Link
              replace={false}
              to="/account/register"
            >
              Create an account.
            </Link>
          </div>
        </FormGroup>
      </form>
    </Col>
  </Row>
</Grid>
`;

exports[`Login component snapshot test should match snapshot 1`] = `
<Grid
  bsClass="container"
  className="login"
  componentClass="div"
  fluid={false}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={6}
      smOffset={3}
      xs={12}
    >
      <form>
        <FormGroup
          bsClass="form-group"
          className="align-center"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Username
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="username-input"
            componentClass="input"
            id="username"
            onChange={[Function]}
            placeholder="E.g. JohnDoe123"
            type="text"
            value="testUsername"
          />
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Password
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="password-input"
            componentClass="input"
            id="password"
            onChange={[Function]}
            placeholder="E.g. 124!Jigzx"
            type="password"
            value="testPassword"
          />
          <Button
            active={false}
            block={true}
            bsClass="btn"
            bsStyle="primary"
            className="login-button"
            disabled={false}
            onClick={[Function]}
          >
            Login
          </Button>
          <div
            className="register-text"
          >
            New to My Saved Recipes? 
            <Link
              replace={false}
              to="/account/register"
            >
              Create an account.
            </Link>
          </div>
        </FormGroup>
      </form>
    </Col>
  </Row>
</Grid>
`;
